libraries/chain/backing_store/tests/benchmark_kv.cpp

```diff
@@ -0,0 +1,337 @@
+   } else if (args.operation == "it_key_value" ) {
+      benchmark_it_key_value(args, std::move(kv_context_ptr), start, end, usage_start, usage_end);
+   };
+}
+
+void print_results(cmd_args& args, wall_clock_t& start, wall_clock_t& end, const rusage& usage_start, const rusage& usage_end) {
+   std::string title = "Benchmarked \"" + args.operation + "\" for ";
+
+   if (args.operation == "it_next" || args.operation == "it_key_value") {
+      title += " one time over ";
+   } else {
+      title += std::to_string(num_iterations);
+      title += " times on ";
+      title += std::to_string(work_set_size);
+      title += " keys from ";
+   }
+
+   title += std::to_string(args.num_keys);
+   title += " total keys ";
+
+   if (args.is_rocksdb) {
+      title += "on RocksDb:";
+   } else {
+      title += "on Chainbase:";
+   }
+
+   uint32_t clock_duration_us = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
+   uint32_t user_duration_us = 1000000 * (usage_end.ru_utime.tv_sec - usage_start.ru_utime.tv_sec) + usage_end.ru_utime.tv_usec - usage_start.ru_utime.tv_usec;;
+   uint32_t system_duration_us = 1000000 * (usage_end.ru_stime.tv_sec - usage_start.ru_stime.tv_sec) + usage_end.ru_stime.tv_usec - usage_start.ru_stime.tv_usec;;
+   uint32_t minor_faults = uint32_t(usage_end.ru_minflt  - usage_start.ru_minflt);
+   uint32_t major_faults = uint32_t(usage_end.ru_majflt  - usage_start.ru_majflt);
+   uint32_t blocks_in = uint32_t(usage_end.ru_inblock - usage_start.ru_inblock);
+   uint32_t blocks_out = uint32_t(usage_end.ru_oublock - usage_start.ru_oublock);
+
+   std::cout << title << std::endl 
+      << "   clock_duration_us: " << clock_duration_us
+      << ", user_duration_us: " << user_duration_us 
+      << ", system_duration_us: " << system_duration_us 
+      << ", minor_faults: " << minor_faults 
+      << ", major_faults: " << major_faults 
+      << ", blocks_in: " << blocks_in  
+      << ", blocks_out: " << blocks_out 
+      << std::endl << std::endl;
+}
+
+void benchmark(cmd_args& args) {
+   uint32_t clock_duration_us;
+   wall_clock_t start, end;
+   struct rusage usage_start, usage_end;
+
+   if (args.is_rocksdb) {
+      boost::filesystem::remove_all("kvrdb-tmp");  // Use a clean RocksDB
+      boost::filesystem::remove_all(chain::config::default_state_dir_name);
+      b1::chain_kv::database      kv_db{"kvrdb-tmp", true};
+      b1::chain_kv::undo_stack    kv_undo_stack{kv_db, vector<char>{make_rocksdb_undo_prefix()}};
+
+      std::unique_ptr<kv_context> kv_context_ptr = create_kv_rocksdb_context<b1::chain_kv::view, b1::chain_kv::write_session, mock_resource_manager>(kv_db, kv_undo_stack, receiver, resource_manager, limits); 
+      benchmark_operation(args, std::move(kv_context_ptr), start, end, usage_start, usage_end); // kv_context_ptr must be in the same scope as kv_db and usage_start, since they are references in create_kv_rocksdb_context
```

Comment:
This also happens in either condition, could we move this after the conditional?

Reply:
benchmark_operation uses kv_context_ptr which requires kv_db or chainbase_db to be in the scope. If we move benchmark_operation after the condition, we would need to move kv_db and chainbase_db before the condition and
initialize both even though only one is needed.